electrum/daemon.py

```diff
@@ -540,6 +546,7 @@ async def _stop_wallet(self, path: str) -> bool:
         wallet = self._wallets.pop(wallet_key, None)
         if not wallet:
             return False
+        self.update_current_wallet()
```

Comment:
One consequence of this flow is that if the GUI program segfaults or gets SIGKILLED at a time when there are multiple wallets open, config.CURRENT_WALLET is left as None and on next launch we try to open `default_wallet`.

Previously in such a case the GUI opened to a reasonable "last" wallet.

Note that in case of a SIGINT, as the program can do a  graceful shutdown, the highlighted line gets a chance to run for each open wallet, and hence config.CURRENT_WALLET will get set to the last one being stopped - which is reasonable, though it might differ from the previous behaviour of "last opened".

Reply:
> One consequence of this flow is that if the GUI program segfaults or gets SIGKILLED at a time when there are multiple wallets open, config.CURRENT_WALLET is left as None and on next launch we try to open default_wallet.

This was fixed in https://github.com/spesmilo/electrum/commit/972ebb04203534b34ad1121358b25f767893cce2